src/player.cpp

```diff
@@ -4835,3 +4961,39 @@ void Player::updateRegeneration()
 		condition->setParam(CONDITION_PARAM_MANATICKS, vocation->getManaGainTicks() * 1000);
 	}
 }
+
+bool Player::addLoyalty(int32_t points)
+{
+	if (loyaltyPoints == 0) {
+		return false;
+	}
+
+	loyaltyPoints += points;
+	return true;
+}
+
+std::string Player::getLoyaltyTitleDescription() const
```

Comment:
this should be a "tool" method, so place this in tools.cpp and pass loyaltyPoints, you can also put everything into a map and just iterate it to find proper title (code will be shorter and easier to edit)